taca/illumina/HiSeq_Runs.py

```diff
@@ -92,138 +92,155 @@ def compute_undetermined(self):
+            #now geenrate the base masks per lane
+            per_lane_base_masks = self._generate_per_lane_base_mask()
+            #store here only the NoIndex lanes
+            per_lane_base_masks_NoIndex = {}
+            run_with_no_index = False # use this flag to check that we are not in the C.Daub case
+            for NoIndexLane in NoIndexLanes:
+                per_lane_base_masks_NoIndex[NoIndexLane] = per_lane_base_masks[NoIndexLane]
+                base_mask_key = per_lane_base_masks[NoIndexLane].keys()[0]
+                new_base_mask = []
+                if len(per_lane_base_masks_NoIndex[NoIndexLane][base_mask_key]['base_mask']):
+                    #C.Daub_15_01 case, only one sample per lane and no index at all
+                    run_with_no_index = True
+                else:
                     for baseMask_element in per_lane_base_masks_NoIndex[NoIndexLane][base_mask_key]['base_mask']:
                         if baseMask_element.startswith("Y"):
                             new_base_mask.append(baseMask_element.replace("Y", "N"))
                         elif baseMask_element.startswith("N"):
                             new_base_mask.append(baseMask_element.replace("N", "Y"))
                     per_lane_base_masks_NoIndex[NoIndexLane][base_mask_key]['base_mask'] = new_base_mask
+            if not run_with_no_index:
+                os.makedirs(NoIndex_Undetermiend)
                 command = self._generate_bcl2fastq_command(per_lane_base_masks_NoIndex, True, "NoIndex", mask_short_adapter_reads=True)
                 with chdir(self.run_dir):
                     misc.call_external_command_detached(command, with_log_files=True, prefix="demux_NoIndex")
                 #return false, as I need to wait to finish the demux for the NoIndex case
                 return False
             else:
-                #in this case it means that I have already started to demux the NoIndex
-                if not os.path.exists(os.path.join(self.run_dir, "Demultiplexing_NoIndex", 'Stats', 'DemultiplexingStats.xml')):
-                    #demultiplexing of undetermined is still ongoing
-                    logger.info("Demux of NoIndex lanes ongoing")
+                #in this case I do not want to start a demux for th eindex, beceause I do not have the index at all
+                #I need to softlink everythin else that is in Stats as I do not want to recompute it
+                missingStatsFiles = glob.glob(os.path.join(self.run_dir, "Demultiplexing_0", "Stats", "*F*L*.txt" ))
+                destination       = os.path.join(self.run_dir, self.demux_dir, "Stats")
+                for source in missingStatsFiles:
+                    source_file_name = os.path.basename(source)
+                    if not os.path.exists(os.path.join(destination, source_file_name)):
+                        os.symlink(source, os.path.join(destination, source_file_name))
```

Comment:
No it will not as the rsync command that is used copies the softlinks as real files. Right now this creates a big data duplication but this is exactly what happens now with illumina_finished_msg

In the future I want to avoid transferring the Demultiplexing_* folders. But before this I want to check that things are working properly